tests/webtests/EntityCreateOidcngTest.php

```diff
@@ -38,6 +38,8 @@ public function setUp()
 
     public function test_it_renders_the_form()
     {
+        // Manage is queried for resource servers
+        $this->testMockHandler->append(new Response(200, [], '[]'));
```

Comment:
Is this boyscouting or is this really needed to support OidcRs?

Reply:
Same for all changes in this file....

Reply:
I notice that sometimes one request is needed and sometimes multiple. Also some lines are documented and some are not so it's not getting any clearer at the moment. And when an empty array with status 200 is expected in multiple different cases how valuable are the functional tests at the moment? Wouldn't it be better to fix it a.s.a.p. as they are currently not representative enough and would have a negative impact on the quality. I think we should think about when to address this.

Reply:
I saw that there was already a task on the drafts cleanup story to address this when cleaning the draft logic. I have created a new story for this to keep the changes more atomic.

[Drafts cleanup story](https://www.pivotaltracker.com/story/show/166004131)
[New fix mock requests story](https://www.pivotaltracker.com/story/show/168513312)

Reply:
As for your first question:

> ... how valuable are the functional tests at the moment? ...

At the time we made the decision to NOT test Manage. So everywhere we can, we simply return the response of the least effort, yielding the desired functional response. The dedicated manage client tests should cover the correct working of that data layer. The throwback from this solution is that the web tests are rather fragile, and it's not always clear when/what is expected to be retrieved from Manage.

My proposal is to not change this right away, but make it part of the bigger refactor effort we have planned (removal of the database layer). If you feel we should address this now, we must certainly plan this with the team. As this would be quite an effort. And should be preceded by a well thought out plan.